### PR TITLE
AudioEditor

### PR DESCRIPTION
**Pull request by Pear-231**
Almost done so submitting a PR :)

### CODE REVIEW

File: Editors/Audio/Presentation/AudioEditor/AudioEditorSettings.cs [hunk 1]
```diff
@@ -0,0 +1,295 @@
+﻿using System.Collections.Generic;
+
+namespace Editors.Audio.Presentation.AudioEditor
+{
+    public class AudioEditorSettings
```
Comment (donkeyProgramming) [inline Editors/Audio/Presentation/AudioEditor/AudioEditorSettings.cs:5]: This class should probably be a static? 
Also, the attributes should only have a get, not a set?

File: Editors/Audio/Presentation/AudioEditor/AudioEditorViewModelHelpers.cs [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+using Editors.Audio.Presentation.AudioEditor.ViewModels;
+using Editors.Audio.Storage;
+using static Editors.Audio.Presentation.AudioEditor.DynamicDataGrid;
+using static Editors.Audio.Presentation.AudioEditor.ViewModels.AudioEditorViewModel;
+
+namespace Editors.Audio.Presentation.AudioEditor
+{
+    public class AudioEditorViewModelHelpers
+    {
+        public static Dictionary<string, List<string>> DialogueEventsWithStateGroupsWithQualifiers { get; set; } = [];
```
Comment (donkeyProgramming) [inline Editors/Audio/Presentation/AudioEditor/AudioEditorViewModelHelpers.cs:12]: Why is this static? That makes it shared across all instances. Should the whole class be static?

File: Editors/Audio/Presentation/AudioEditor/AudioEditorViewModelHelpers.cs [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+                EventsData[dialogueEvent] = dataGridItems;
+            }
+        }
+
+        public static void LoadEvent(AudioEditorViewModel viewModel, IAudioRepository audioRepository, bool showCustomStatesOnly)
+        {
+            if (string.IsNullOrEmpty(AudioEditorData.Instance.SelectedAudioProjectEvent))
+                return;
+
+            Debug.WriteLine($"Loading event: {AudioEditorData.Instance.SelectedAudioProjectEvent}");
```
Comment (donkeyProgramming) [inline Editors/Audio/Presentation/AudioEditor/AudioEditorViewModelHelpers.cs:44]: Either use the logger system or remove the debug print.

File: Editors/Audio/Presentation/AudioEditor/AudioProjectData.cs [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+
+            var settings = new JObject
+            {
+                ["BnkName"] = "battle_vo_conversational__ovn_vo_actor_Albion_Dural_Durak",
+                ["Language"] = "english(uk)"
+            };
+
+            audioProject["Settings"] = settings;
+
+            var dialogueEvents = new JArray();
```
Comment (donkeyProgramming) [inline Editors/Audio/Presentation/AudioEditor/AudioProjectData.cs:59]: Do not use the Jarray and other related json based data sources directly. They generally only make problems in life

Comment (donkeyProgramming) [reply Editors/Audio/Presentation/AudioEditor/AudioProjectData.cs:59]: https://learn.microsoft.com/en-us/dotnet/standard/serialization/system-text-json/how-to

File: Editors/Audio/Presentation/AudioEditor/AudioProjectData.cs [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+
+                dialogueEvents.Add(dialogueEvent);
+            }
+
+            audioProject["DialogueEvents"] = dialogueEvents;
+
+            return audioProject.ToString(Formatting.Indented);
+        }
+
+        public static Dictionary<string, List<Dictionary<string, object>>> ConvertAudioProjectToEventData(IAudioRepository audioRepository, string audioProjectJson)
```
Comment (donkeyProgramming) [inline Editors/Audio/Presentation/AudioEditor/AudioProjectData.cs:100]: https://learn.microsoft.com/en-us/dotnet/standard/serialization/system-text-json/how-to

File: Editors/Audio/Presentation/AudioEditor/ViewModels/AudioEditorViewModel.cs [hunk 6]
```diff
@@ -0,0 +1,291 @@
...
+        Custom
+    }
+
+    public partial class AudioEditorViewModel : ObservableObject, IEditorViewModel
+    {
+        public readonly IAudioRepository AudioRepository;
+        public readonly PackFileService PackFileService;
+
+        public string _previousSelectedAudioProjectEvent;
+        [ObservableProperty]
```
Comment (donkeyProgramming) [inline Editors/Audio/Presentation/AudioEditor/ViewModels/AudioEditorViewModel.cs:44]: preference, but I think it looks very neat when you do this as one line
[observableProp] float _someAttribute

File: Editors/Audio/Presentation/AudioEditor/ViewModels/AudioEditorViewModel.cs [hunk 7]
```diff
@@ -0,0 +1,291 @@
...
+        // Audio Project settings
+        [ObservableProperty]
+        private string _selectedAudioProjectEventType;
+        [ObservableProperty]
+        private string _selectedAudioProjectEventSubtype;
+        [ObservableProperty]
+        private DialogueEventsPreset _selectedAudioProjectEventsPreset;
+
+        public NotifyAttr<string> DisplayName { get; set; } = new NotifyAttr<string>("Audio Editor");
+        public AudioEditorDataViewModel AudioEditor { get; }
```
Comment (donkeyProgramming) [inline Editors/Audio/Presentation/AudioEditor/ViewModels/AudioEditorViewModel.cs:58]: There is a lot of public attributes here. This makes me suspect that data is acessed pleases where it should not be? 
Or that it should use the wpf community toolkit => [ObservableProperty]

File: Editors/Audio/Presentation/AudioEditor/ViewModels/AudioEditorViewModel.cs [hunk 8]
```diff
@@ -0,0 +1,291 @@
...
+        public ICommand CreateAudioProjectCommand { get; set; }
+
+        public AudioEditorViewModel(AudioEditorDataViewModel audioEditor, AudioEditorSettingsViewModel audioEditorSettings, IAudioRepository audioRepository, PackFileService packFileService)
+        {
+            AudioEditor = audioEditor;
+            AudioEditorSettings = audioEditorSettings;
+            AudioRepository = audioRepository;
+            PackFileService = packFileService;
+
+            AddStatePathCommand = new RelayCommand(AddStatePath);
```
Comment (donkeyProgramming) [inline Editors/Audio/Presentation/AudioEditor/ViewModels/AudioEditorViewModel.cs:85]: Relay commands can also be cleaned up using the wpf toolkit

**Review by donkeyProgramming — CHANGES_REQUESTED** [review]
Lots of code, so only did a quick reveiw. Pointed out some minor issues which probably was mistakes

File: Editors/Audio/Presentation/AudioEditor/ViewModels/AudioEditorViewModel.cs (hunk 6 above)
Comment (Pear-231) [reply Editors/Audio/Presentation/AudioEditor/ViewModels/AudioEditorViewModel.cs:44]: Didn't even think of that, nice will do.

File: Editors/Audio/Presentation/AudioEditor/ViewModels/AudioEditorViewModel.cs (hunk 7 above)
Comment (Pear-231) [reply Editors/Audio/Presentation/AudioEditor/ViewModels/AudioEditorViewModel.cs:58]: I'll look over it :)

File: Editors/Audio/Presentation/AudioEditor/ViewModels/AudioEditorViewModel.cs (hunk 8 above)
Comment (Pear-231) [reply Editors/Audio/Presentation/AudioEditor/ViewModels/AudioEditorViewModel.cs:85]: Oh looks like I missed those ones, good point!

File: Editors/Audio/Presentation/AudioEditor/AudioProjectData.cs (hunk 4 above)
Comment (Pear-231) [reply Editors/Audio/Presentation/AudioEditor/AudioProjectData.cs:59]: Interesting. Will take a look at this.

**Pull request merged**